### PR TITLE
[AAE-12124] Use a function to remove js files

### PR DESCRIPTION
**Pull request by mauriziovitale**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

**Review by arditdomi — APPROVED** [review]

**Review by arditdomi — APPROVED** [review]

File: lib/config/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+function DisableOutputWebpackPlugin(options) {
+	if (options && options.test && !Array.isArray(options.test))
+		options.test = [options.test]
```
Comment (ehsan-2019) [inline lib/config/index.js:3]: you can consider to put this line into a curly braces.

File: lib/config/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			}
+			Object.keys(compilation.assets).forEach((asset) => {
+				let output = true
+				this.options.test.some((regex) => {
+					if (asset.match(regex) != null) {
+						output = false
+						return true
+					}
+					return false
+				})
```
Comment (ehsan-2019) [inline lib/config/index.js:23]: you can consider to refactor forEach block like this:
`	const output = this.options.test.some(regex => !!asset.match(regex));`

	if (!output) {
		delete compilation.assets[asset];
	}

File: lib/config/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+				if (!output)
+					delete compilation.assets[asset]
+			});
+		} else {
+			Object.keys(compilation.assets).forEach((asset) => {
+				delete compilation.assets[asset]
```
Comment (ehsan-2019) [inline lib/config/index.js:30]: missing semicolon

File: lib/config/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	this.options = options
+}
+
+DisableOutputWebpackPlugin.prototype.apply = function(compiler) {
+	compiler.hooks.emit.tapAsync('DisableOutputWebpackPlugin', (compilation, callback) => {
+
+		if (this.options && this.options.test) {
+			if (Object.keys(compilation.assets).length === 0 ) {
+				throw Error ('Error: The asset pre-theme is not there!')
```
Comment (ehsan-2019) [inline lib/config/index.js:13]: missing semicolon

File: lib/config/index.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+function DisableOutputWebpackPlugin(options) {
+	if (options && options.test && !Array.isArray(options.test))
+		options.test = [options.test]
+
+	this.options = options
```
Comment (ehsan-2019) [inline lib/config/index.js:5]: missing semicolon

**Review by ehsan-2019 — CHANGES_REQUESTED** [review]

**Review by ehsan-2019 — APPROVED** [review]
I approve the PR. comments are optional.

**Pull request merged**